runtime/tests/jniarg/args_10.c

```diff
@@ -0,0 +1,184 @@
+
+/*******************************************************************************
+ * Copyright (c) 2019, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "jniargtests.h"
+
+jboolean JNICALL Java_JniArgTests_nativeZIZIZIZIZIrZ(JNIEnv *p_env, jobject p_this, int arg1, int arg2, int arg3, int arg4, int arg5, int arg6, int arg7, int arg8, int arg9, int arg10)
```

Comment:
`int` should that be `jint`?  and `jboolean`?  The `ZIZIZIZIZI` signature doesn't match the declared types of the args

Reply:
@DanHeidinga ,  the behavior of test cases are based on the code changes in this PR.

Given that changing `uint8 `(1byte) to `uint32 `(4bytes) for boolean type affects both arguments and return type in size, we need to make sure the following cases are correct with my fix:
e.g.
```
definition in java                                             declaration/implementation in the native
1) nativeZIrZ(int arg1, boolean arg2) --->  nativeZIrZ(jint arg1, jint arg2)  (extend boolean to int in the native, which is exactly the issue we fixed in this PR)
2) nativeZIrZ(int arg1, boolean arg2) --->  nativeZIrZ(jint arg1, jboolean arg2)  (remains boolean  in the native to ensure it still works good)
3) return boolean in different ways (directly TRUE/FALSE or indirectly with == statement which only set value to AL in x86) to ensure it returns correctly on both little-endian and big-endian.
```

I will added the test case you mentioned above (which I overlooked) to see whether it works  good.

Reply:
@DanHeidinga ,  the behavior of test cases are based on the code changes in this PR.

Given that changing `uint8 `(1byte) to `uint32 `(4bytes) for boolean type affects both arguments and return type in size, we need to make sure the following cases are correct with my fix:
e.g.
```
definition in java                                             declaration/implementation in the native
1) nativeZIrZ(int arg1, boolean arg2) --->  nativeZIrZ(jint arg1, jint arg2)  (extend boolean to int in the native, which is exactly the issue we fixed in this PR)
2) nativeZIrZ(int arg1, boolean arg2) --->  nativeZIrZ(jint arg1, jboolean arg2)  (remains boolean  in the native to ensure it still works good)
3) return boolean in different ways (directly TRUE/FALSE or indirectly with == statement which only set value to AL in x86) to ensure it returns correctly on both little-endian and big-endian.
```

I will added the test case you mentioned above (which I overlooked) to see whether it works  good.